libpimeval/src/pimSim.cpp

```diff
@@ -55,6 +57,32 @@ pimSim::~pimSim()
   delete m_paramsPerf;
 }
 
+//! @brief  Initialize pimSim member classes from the config file
+void
+pimSim::init(const std::string& simConfig)
+{
+  bool success = parseConfigFromFile(simConfig);
+  assert(success);
+
+  std::string memConfigFileFullPath = m_configFilesPath + m_memConfigFileName;
+  std::string fileContent;
+  success = pimUtils::readFileContent(memConfigFileFullPath.c_str(), fileContent);
+  assert(success);
+
+  m_paramsDram = new pimParamsDram(fileContent);
+  m_paramsPerf = new pimParamsPerf(m_paramsDram);
+  m_statsMgr = new pimStatsMgr(m_paramsDram, m_paramsPerf);
+}
+
+//! @brief  Initialize pimSim member classes with default values
+void
+pimSim::init()
```

Comment:
We need either uninit or handle the case of calling init twice and avoid MLK

Reply:
@deyuan
Thanks for catching this, MLK. To resolve this issue, I propose adding a new private member field to the pimSim class to track whether init has been called. If init has already been called, an error status should be returned. Do you have any other suggestions?